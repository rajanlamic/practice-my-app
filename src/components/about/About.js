import React from "react";
import { Button } from "reactstrap";
import {
  setDevelopment,
  setEnhancement,
  asynchronousDevelopment,
  loadCategoires
} from "./about-actions";
import { connect } from "react-redux";
import AboutChild from "./AboutChild";

export class About extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      myVal: "myValue original"
    };
    this.clickHandler = this.clickHandler.bind(this);
  }

  clickHandler() {
    this.setState(prevState => {
      return {
        myVal: `${prevState.myVal} myValue updated value`
      };
    });
  }

  clickHandler1 = () => {
    this.setState({
      myVal: `myValue second updated value`
    });
  };

  reduxHandlerDevelopment = () => {
    this.props.onDevelopment();
  };

  reduxHandlerEnhancement = () => {
    this.props.onEnhancement();
  };

  asynchronousDevelopmentClick = () => {
    this.props.onAsynchronousDevelopment();
  };

  loadCategoriesHandler = () => {
    this.props.onLoadCategories();
  };
  render() {
    return (
      <div>
        <h2>About</h2>
        <Button onClick={this.clickHandler}>Click</Button>
        <Button onClick={this.clickHandler1}>Click1</Button>
        <div>{this.state.myVal}</div>
        <Button onClick={this.reduxHandlerDevelopment}>
          Redux Click Development
        </Button>
        {this.props.isDevelopment && (
          <div className="development">Development Clicked</div>
        )}
        <Button onClick={this.reduxHandlerEnhancement}>
          Redux Click Enhancement
        </Button>
        {this.props.isEnhancement && (
          <div className="enhancement">Enhancement Clicked</div>
        )}
        <AboutChild myProp="myProp" />
        <Button onClick={this.asynchronousDevelopmentClick}>
          Asynchronous Click Development
        </Button>
        {this.props.isAsyncDevelopment && (
          <div className="async-development">Async Development Clicked</div>
        )}
        <Button onClick={this.loadCategoriesHandler}>Load Categories</Button>
        {this.props.isFetched &&
          this.props.data &&
          this.props.data.children &&
          this.props.data.children.map(item => {
            return <li>{item.title}</li>;
          })}
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    isDevelopment: state.about.isDevelopment,
    isEnhancement: state.about.isEnhancement,
    isAsyncDevelopment: state.about.isAsyncDevelopment,
    isFetched: state.about.isFetched,
    data: state.about.data
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onDevelopment: () => {
      dispatch(setDevelopment());
    },
    onEnhancement: () => {
      dispatch(setEnhancement());
    },
    onAsynchronousDevelopment: () => {
      dispatch(asynchronousDevelopment());
    },
    onLoadCategories: () => {
      dispatch(loadCategoires());
    }
  };
};
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(About);
