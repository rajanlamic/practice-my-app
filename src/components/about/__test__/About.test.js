import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });
import { About } from "../About";

describe("About", () => {
  let wrapper, props;
  beforeEach(() => {
    props = {
      isDevelopment: true,
      isEnhancement: false
    };
    wrapper = shallow(<About {...props} />);
  });
  afterEach(() => {
    wrapper = null;
  });

  it("should render About properly with snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render development if development props is passed", () => {
    expect(wrapper.find(".development").length).toEqual(1);
  });
  it("should have myProp passed to AboutChild", () => {
    let aboutChild = wrapper.find("AboutChild");
    expect(aboutChild.props().myProp).toEqual("myProp");
  });
});
