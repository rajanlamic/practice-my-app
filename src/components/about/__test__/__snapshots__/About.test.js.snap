// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should render About properly with snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <About
    isDevelopment={true}
    isEnhancement={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          About
</h2>,
        <Button
          color="secondary"
          onClick={[Function]}
          tag="button"
>
          Click
</Button>,
        <Button
          color="secondary"
          onClick={[Function]}
          tag="button"
>
          Click1
</Button>,
        <div>
          myValue original
</div>,
        <Button
          color="secondary"
          onClick={[Function]}
          tag="button"
>
          Redux Click Development
</Button>,
        <div
          className="development"
>
          Development Clicked
</div>,
        <Button
          color="secondary"
          onClick={[Function]}
          tag="button"
>
          Redux Click Enhancement
</Button>,
        false,
        <AboutChild
          myProp="myProp"
/>,
        <Button
          color="secondary"
          onClick={[Function]}
          tag="button"
>
          Asynchronous Click Development
</Button>,
        undefined,
        <Button
          color="secondary"
          onClick={[Function]}
          tag="button"
>
          Load Categories
</Button>,
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "About",
        },
        "ref": null,
        "rendered": "About",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Click",
          "color": "secondary",
          "onClick": [Function],
          "tag": "button",
        },
        "ref": null,
        "rendered": "Click",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Click1",
          "color": "secondary",
          "onClick": [Function],
          "tag": "button",
        },
        "ref": null,
        "rendered": "Click1",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "myValue original",
        },
        "ref": null,
        "rendered": "myValue original",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Redux Click Development",
          "color": "secondary",
          "onClick": [Function],
          "tag": "button",
        },
        "ref": null,
        "rendered": "Redux Click Development",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Development Clicked",
          "className": "development",
        },
        "ref": null,
        "rendered": "Development Clicked",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Redux Click Enhancement",
          "color": "secondary",
          "onClick": [Function],
          "tag": "button",
        },
        "ref": null,
        "rendered": "Redux Click Enhancement",
        "type": [Function],
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "myProp": "myProp",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Asynchronous Click Development",
          "color": "secondary",
          "onClick": [Function],
          "tag": "button",
        },
        "ref": null,
        "rendered": "Asynchronous Click Development",
        "type": [Function],
      },
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Load Categories",
          "color": "secondary",
          "onClick": [Function],
          "tag": "button",
        },
        "ref": null,
        "rendered": "Load Categories",
        "type": [Function],
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            About
</h2>,
          <Button
            color="secondary"
            onClick={[Function]}
            tag="button"
>
            Click
</Button>,
          <Button
            color="secondary"
            onClick={[Function]}
            tag="button"
>
            Click1
</Button>,
          <div>
            myValue original
</div>,
          <Button
            color="secondary"
            onClick={[Function]}
            tag="button"
>
            Redux Click Development
</Button>,
          <div
            className="development"
>
            Development Clicked
</div>,
          <Button
            color="secondary"
            onClick={[Function]}
            tag="button"
>
            Redux Click Enhancement
</Button>,
          false,
          <AboutChild
            myProp="myProp"
/>,
          <Button
            color="secondary"
            onClick={[Function]}
            tag="button"
>
            Asynchronous Click Development
</Button>,
          undefined,
          <Button
            color="secondary"
            onClick={[Function]}
            tag="button"
>
            Load Categories
</Button>,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "About",
          },
          "ref": null,
          "rendered": "About",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Click",
            "color": "secondary",
            "onClick": [Function],
            "tag": "button",
          },
          "ref": null,
          "rendered": "Click",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Click1",
            "color": "secondary",
            "onClick": [Function],
            "tag": "button",
          },
          "ref": null,
          "rendered": "Click1",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "myValue original",
          },
          "ref": null,
          "rendered": "myValue original",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Redux Click Development",
            "color": "secondary",
            "onClick": [Function],
            "tag": "button",
          },
          "ref": null,
          "rendered": "Redux Click Development",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Development Clicked",
            "className": "development",
          },
          "ref": null,
          "rendered": "Development Clicked",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Redux Click Enhancement",
            "color": "secondary",
            "onClick": [Function],
            "tag": "button",
          },
          "ref": null,
          "rendered": "Redux Click Enhancement",
          "type": [Function],
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "myProp": "myProp",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Asynchronous Click Development",
            "color": "secondary",
            "onClick": [Function],
            "tag": "button",
          },
          "ref": null,
          "rendered": "Asynchronous Click Development",
          "type": [Function],
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Load Categories",
            "color": "secondary",
            "onClick": [Function],
            "tag": "button",
          },
          "ref": null,
          "rendered": "Load Categories",
          "type": [Function],
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
